spec/models/state_file_ny_intake_spec.rb

```diff
@@ -0,0 +1,126 @@
+#  public_housing                   :integer          default("unfilled"), not null
+#  raw_direct_file_data             :text
+#  referrer                         :string
+#  refund_choice                    :integer          default("unfilled"), not null
+#  residence_county                 :string
+#  routing_number                   :string
+#  sales_use_tax                    :integer
+#  sales_use_tax_calculation_method :integer          default("unfilled"), not null
+#  school_district                  :string
+#  school_district_number           :integer
+#  source                           :string
+#  spouse_birth_date                :date
+#  spouse_first_name                :string
+#  spouse_last_name                 :string
+#  spouse_middle_initial            :string
+#  spouse_signature                 :string
+#  untaxed_out_of_state_purchases   :integer          default("unfilled"), not null
+#  created_at                       :datetime         not null
+#  updated_at                       :datetime         not null
+#  visitor_id                       :string
+#
+require "rails_helper"
+
+describe StateFileNyIntake do
+  describe "before_save" do
+    let(:intake) { create :state_file_ny_intake, untaxed_out_of_state_purchases: "yes", sales_use_tax_calculation_method: "manual", sales_use_tax: "350", household_fed_agi: 102_000 }
+
+    context "when untaxed_out_of_state_purchases changes to no" do
+      it "clears sales use tax calculation method and sales use tax" do
+        expect {
+          intake.update(untaxed_out_of_state_purchases: "no")
+        }.to change(intake, :sales_use_tax_calculation_method).to("unfilled")
+         .and change(intake, :sales_use_tax).to(nil)
+      end
+    end
+
+    context "when sales_use_tax_calculation_method changes to automated" do
+      it "calculates sales use tax" do
+        expect {
+          intake.update(sales_use_tax_calculation_method: "automated")
+        }.to change(intake, :sales_use_tax).to(29)
+      end
+    end
+
+  end
+
+  describe "#calculate_sales_use_tax" do
+    let(:intake) { build :state_file_ny_intake, household_fed_agi: household_fed_agi }
+    let(:household_fed_agi) { 14_000 }
+
+    context "when the federal agi is 14,000" do
+      it "returns 3" do
+        expect(intake.calculate_sales_use_tax).to eq 3
+      end
+    end
+
+    context "when the federal agi is 75,001" do
+      let(:household_fed_agi) { 75_001 }
+      it "returns 23" do
+        expect(intake.calculate_sales_use_tax).to eq 23
+      end
+    end
+
+    context "when the federal agi is 201,000" do
+      let(:household_fed_agi) { 201_000 }
+      it "returns calculation" do
+        expect(intake.calculate_sales_use_tax).to eq (201_000 * 0.000195).round
+      end
+    end
+
+    context "when the federal agi is 700,000" do
+      let(:household_fed_agi) { 700_000 }
+      it "returns 125" do
+        expect(intake.calculate_sales_use_tax).to eq 125
+      end
+    end
+  end
+end
```

Comment:
This and the method in the model are super clear to me as a reviewer. This is very readable!

Reply:
This could maybe use a test for the escape clause (no fed agi) but that doesn't feel critical to me.